rust/compiler_builtins.rs

```diff
@@ -51,6 +52,24 @@ define_panicking_intrinsics!("`u128` should not be used", {
 });
 
 #[cfg(target_arch = "arm")]
-define_panicking_intrinsics!("`u64` division/modulo should not be used", {
-    __aeabi_uldivmod,
-});
+#[doc(hidden)]
+#[naked]
+#[no_mangle]
+pub extern "C" fn rust_aeabi_uldivmod() {
```

Comment:
I believe you still have to use the correct function signature even for `#[naked]` functions. In addition the calling convention needs to match what is actually used. Finally you don't need `pub` here. `#[no_mangle]` is enough to export the symbol.

Reply:
The function signature for `#[naked]` function doesn't matter for the assembly. It's only a description that how Rust code should call it.

Reply:
Seems like compiler-builtins doesn't specify the parameters either due to them having a custom calling convention. It does mark them as `unsafe` though.